src/mqt/benchviewer/backend.py

```diff
@@ -23,256 +23,544 @@
+                                print(
+                                    "Found 'MQTBench_all.zip' (Version {}, Size {} MB, Link: {})".format(
+                                        possible_version,
+                                        file_size,
+                                        download_url,
+                                    )
+                                )
+                                response = input("Would you like to downloaded the file? (Y/n)")
+                            if skip_question or response.lower() == "y" or response == "":
+                                self.handle_downloading_benchmarks(target_location, download_url)
+                                break
+                if version_found:
+                    break
+
+            if not version_found:
+                print("No suitable benchmarks found.")
+                return False
+
+        with huge_zip_path.open("rb") as zf:
+            zip_bytes = io.BytesIO(zf.read())
+            self.mqtbench_all_zip = ZipFile(zip_bytes, mode="r")
+        return True
 
+    def handle_downloading_benchmarks(self, target_location: str, download_url: str):
+        print("Start downloading benchmarks...")
 
-def handle_downloading_benchmarks(target_location: str, download_url: str):
-    print("Start downloading benchmarks...")
+        r = requests.get(download_url)
+        # r = self.handle_github_api_request(download_url)
+        total_length = r.headers.get("content-length")
+        total_length = int(total_length)
+        fname = target_location + "/MQTBench_all.zip"
 
-    r = requests.get(download_url)
-    total_length = r.headers.get("content-length")
-    total_length = int(total_length)
-    fname = target_location + "/MQTBench_all.zip"
+        Path(target_location).mkdir(parents=True, exist_ok=True)
+        with Path(fname).open("wb") as f, tqdm(
+            desc=fname,
+            total=total_length,
+            unit="iB",
+            unit_scale=True,
+            unit_divisor=1024,
+        ) as bar:
+            for data in r.iter_content(chunk_size=1024):
+                size = f.write(data)
+                bar.update(size)
+        print(f"Download completed to {fname}. Server is starting now.")
 
-    Path(target_location).mkdir(parents=True, exist_ok=True)
-    with Path(fname).open("wb") as f, tqdm(
-        desc=fname,
-        total=total_length,
-        unit="iB",
-        unit_scale=True,
-        unit_divisor=1024,
-    ) as bar:
-        for data in r.iter_content(chunk_size=1024):
-            size = f.write(data)
-            bar.update(size)
-    print(f"Download completed to {fname}. Server is starting now.")
+        return
 
 
 def get_tket_settings(filename: str):
     if "line" in filename:
         return "line"
     if "graph" in filename:
         return "graph"
-    return None
+    return ValueError("Unknown tket_setting: " + filename)
```

Comment:
why is an error _returned_ here instead of being **raised**?
Please double check any other places where that might have happened.

Reply:
Fixed that. Before I thought `None` marks the erroneous case but this is not true, since for the independent level files still `None` should be returned, since no tket settings can be set.